BlueprintUI/Sources/Layout/EqualStack.swift

```diff
@@ -153,6 +153,75 @@ extension EqualStack {
             return result
         }
 
+        func sizeThatFits(proposal: SizeConstraint, subelements: Subelements, cache: inout Cache) -> CGSize {
+            guard subelements.count > 0 else { return .zero }
+
+            let totalSpacing = (spacing * CGFloat(subelements.count - 1))
+            let itemProposal: SizeConstraint
+            switch direction {
+            case .horizontal:
+                itemProposal = .init(
+                    width: proposal.width.map { ($0 - totalSpacing) / CGFloat(subelements.count) },
+                    height: proposal.height
+                )
+            case .vertical:
+                itemProposal = .init(
+                    width: proposal.width,
+                    height: proposal.height.map { ($0 - totalSpacing) / CGFloat(subelements.count) }
+                )
+            }
+            let itemSizes = subelements.map { $0.sizeThatFits(itemProposal) }
+
+            let maximumItemWidth = itemSizes.map { $0.width }.max() ?? 0
+            let maximumItemHeight = itemSizes.map { $0.height }.max() ?? 0
+
+            let totalSize: CGSize
+            switch direction {
+            case .horizontal:
+                totalSize = CGSize(
+                    width: maximumItemWidth * CGFloat(subelements.count) + spacing * CGFloat(subelements.count - 1),
+                    height: maximumItemHeight
+                )
+            case .vertical:
+                totalSize = CGSize(
+                    width: maximumItemWidth,
+                    height: maximumItemHeight * CGFloat(subelements.count) + spacing * CGFloat(subelements.count - 1)
+                )
+            }
+
+            return totalSize
+        }
+
+        func placeSubelements(in size: CGSize, subelements: Subelements, cache: inout ()) {
+            guard subelements.count > 0 else { return }
+
+            let totalSpacing = (spacing * CGFloat(subelements.count - 1))
+            let itemSize: CGSize
+            switch direction {
+            case .horizontal:
+                itemSize = CGSize(
+                    width: (size.width - totalSpacing) / CGFloat(subelements.count),
+                    height: size.height
+                )
+            case .vertical:
+                itemSize = CGSize(
+                    width: size.width,
+                    height: (size.height - totalSpacing) / CGFloat(subelements.count)
+                )
+            }
+
+            for (subelement, index) in zip(subelements, 0...) {
```

Comment:
Noting: As we've seen in the past, might be faster to use `indexedForEach` here.

Reply:
We've seen that any use of zip or enumerated is slower than our indexed helpers, likely due to allocations happening (even if on the stack) and extra internal overhead.

Reply:
Interesting! I wasn't seeing any significant differences in my quick profiling (release mode on Mac) of these examples:

<details>

```
let values = repeatElement("Hello", count: 10_000_000_000)

var stored: (String, Int) = ("", -1)

func testZip() {
    for (item, offset) in zip(values, 0...) {
        stored = (item, offset)
    }
}

func testForIn() {
    for item in values {
        stored = (item, -1)
    }
}

func testEnumerated() {
    for (offset, item) in values.enumerated() {
        stored = (item, offset)
    }
}

//testZip() // 22.03s
//testForIn() // 22.04s
testEnumerated() // 22.06s
```

</details>

[enumerated-vs-zip-vs-forIn.trace.zip](https://github.com/square/Blueprint/files/11104102/enumerated-vs-zip-vs-forIn.trace.zip)